misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/FileUtil.java

```diff
@@ -64,4 +72,57 @@ public String stripIllegalCharacters(@Nullable String filename)
 
 		return sb.toString();
 	}
+
+	@NonNull
+	public Path getFilePath(@NonNull final URL url) throws MalformedURLException
+	{
+		final boolean isWindowsLocalPath = Check.isEmpty(url.getHost())
+				? url.getPath().contains(":")
+				: url.getAuthority().contains(":");
+
+		final Path path = isWindowsLocalPath
+				? parseWindowsLocalPath(url)
+				: parseFileURL(url);
+
+		if (path == null)
+		{
+			throw new RuntimeException("Couldn't parse path from:" + url);
+		}
+
+		return path;
+	}
+
+	@Nullable
+	private Path parseWindowsLocalPath(@NonNull final URL url)
```

Comment:
why Windows? What's specific to Windows?

btw, our production code runs on Linux.

Reply:
yeah,,, but we run it on windows so it got in the way :) the "special" thing is the partition e.g. `C:`